contracts/sol6/helpers/KyberRateQueryHelper.sol

```diff
@@ -0,0 +1,151 @@
+pragma solidity 0.6.6;
+
+import "@kyber.network/utils-sc/contracts/Utils.sol";
+import "@kyber.network/utils-sc/contracts/IERC20Ext.sol";
+import "../IConversionRates.sol";
+import "../IKyberSanity.sol";
+import "../IKyberReserve.sol";
+
+interface IKyberReserveExt is IKyberReserve {
+    function conversionRatesContract() external view returns (IConversionRates);
+    function sanityRatesContract() external view returns (IKyberSanity);
+}
+
+contract KyberRateQueryHelper is Utils {
+
+    function getRatesWithEth(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata tokens,
+        uint256 weiAmount
+    )
+        external view returns(uint256[] memory sellRates, uint256[] memory buyRates)
+    {
+        uint256 numTokens = tokens.length;
+
+        buyRates = new uint256[](numTokens);
+        sellRates = new uint256[](numTokens);
+        for (uint256 i = 0; i < numTokens; i++) {
+            (buyRates[i], sellRates[i], , , ) = getRateWithEth(reserve, tokens[i], weiAmount);
+        }
+    }
+
+    function getRatesWithToken(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata tokens,
+        uint256 tweiAmount
+    )
+        external view 
+        returns(uint256[] memory sellRates, uint256[] memory buyRates)
+    {
+        uint256 numTokens = tokens.length;
+
+        buyRates = new uint256[](numTokens);
+        sellRates = new uint256[](numTokens);
+        for (uint256 i = 0; i < numTokens; i++) {
+            (buyRates[i], sellRates[i], ,  , ) = getRateWithEth(reserve, tokens[i], tweiAmount);
+        }
+    }
+
+    function getReserveRates(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata srcs,
+        IERC20Ext[] calldata dests
+    )
+        external view returns(uint256[] memory pricingRates, uint256[] memory sanityRates)
+    {
+        require(srcs.length == dests.length, "srcs length != dests");
+
+        pricingRates = new uint256[](srcs.length);
+        sanityRates = new uint256[](srcs.length);
+
+        for (uint256 i = 0 ; i < srcs.length ; i++) {
+
+            if (reserve.sanityRatesContract() != IKyberSanity(0x0)) {
```

Comment:
it's better to use `try catch` here as some reserves (like bridge reserves) don't have `sanityRatesContract`
similar for `conversionRatesContract`

Reply:
Yes
not sure it matters. due to the way its used. 
but added, just in case.